src/preact/parse-props.js

```diff
@@ -10,6 +10,13 @@ import * as Preact from '#preact';
 
 import {Slot, createSlot} from './slot';
 
+/** @typedef {*} MediaQueryProps */
+
+/** @typedef {Object<string, AmpElementProp>} AmpElementProps */
+
+// TODO: we can create much more accurate typing for AmpElementProps, which will describe
+// how it is meant to be used. This will also allow us to remove various devAsserts.
+
 /**
```

Comment:
potential for the future: we can create much more accurate typing for AmpElementProps, which would describe which sets of properties go together and which are mutually exclusive etc. Then we could also remove various devAssert statements.